smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -193,11 +193,11 @@ private boolean isMemberOverwriteRequired(MemberShape member, Set<String> parent
         }
 
         Shape memberTarget = model.expectShape(member.getTarget());
-        parents.add(symbolProvider.toMemberName(member));
         if (memberTarget.isUnionShape()) {
             // always call filterSensitiveLog for UnionShape
```

Comment:
Unions can also safely be recursive.

Reply:
We return true for `isMemberOverwriteRequired` for Unions, so recursive check never happens when we encounter Unions.